File: pdftract-java/src/main/java/com/jedarden/pdftract/Block.java
package com.jedarden.pdftract;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/**
 * A semantic block (paragraph, heading, table, etc.).
 */
public record Block(
    @JsonProperty("kind") String kind,
    @JsonProperty("bbox") List<Double> bbox,
    @JsonProperty("lines") List<Line> lines
) {
    public Block {
        bbox = bbox != null ? bbox : List.of();
        lines = lines != null ? lines : List.of();
    }
}

File: pdftract-dotnet/src/Pdftract/Models/Block.cs
using System.Text.Json.Serialization;

namespace Pdftract.Models;

/// <summary>
/// Represents a structural block (paragraph, heading, table, etc.).
/// </summary>
public record Block
{
    [JsonPropertyName("kind")]
    public required string Kind { get; init; }

    [JsonPropertyName("text")]
    public required string Text { get; init; }

    [JsonPropertyName("bbox")]
    public required double[] Bbox { get; init; }

    [JsonPropertyName("level")]
    public int? Level { get; init; }
}
